plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/wizard/connection/AdvancedConnectionEditor.java

```diff
@@ -351,11 +351,19 @@ public boolean isViewFor(Object object) {
 	}
 
 	public Map<String, Object> getExtendedProperties() {
+		if( map != null ) {
+			return map;
+		}
 		IConnection connection = pageModel.getSelectedConnection();
-		if (!(connection instanceof Connection)) {
-			return null;
+		if( connection != null ) {
+			if( connection instanceof Connection) {
+				map = ((Connection)connection).getExtendedProperties();
+			} else {
+				map = new HashMap<String, Object>();
```

Comment:
why should we return null if the object is no connection at all while return an empty map when it's a v2 connection? why this distinction?